lib/charms/opensearch/v0/opensearch_base_charm.py

```diff
@@ -1142,6 +1142,10 @@ def _post_start_init(self, event: _StartOpenSearch):  # noqa: C901
         if self.opensearch_peer_cm.is_provider():
             self.peer_cluster_provider.refresh_relation_data(event, can_defer=False)
 
+        # before resetting the CA rotation state, we remove the old ca from the truststore
+        if self.peers_data.get(Scope.UNIT, "tls_ca_renewed", False):
```

Comment:
I think we should instead check on `ca_rotation_complete_in_cluster` to ensure that all nodes / all applications have finished rotating the certificates after a CA rotation, no?

Reply:
But that would never happen at the same time, or we would need to defer the `_StartOpenSearch` event here.